validator/sawtooth_validator/consensus/notifier.py

```diff
@@ -28,20 +29,22 @@ class ConsensusNotifier:
 
     def __init__(self, consensus_service):
         self._service = consensus_service
+        self._registered_engines = ConcurrentSet()
 
     def _notify(self, message_type, message):
-        futures = self._service.send_all(
-            message_type,
-            message.SerializeToString())
-        for future in futures:
-            future.result()
+        if self._registered_engines:
```

Comment:
For this statement to work, you need to implement `__bool__` on `ConcurrentSet` - otherwise this is always true (the default `bool` operation is `None==False`, `!None == True`)

Reply:
When `__bool__` isn't defined, Python calls `__len__` instead: https://docs.python.org/3/reference/datamodel.html#object.__bool__